lib/facter/util/resolvers/ffi/hostname.rb

```diff
@@ -36,19 +36,23 @@ def self.getffihostname
             raw_hostname.read_string
           end
 
-          def self.getffiaddrinfo(hostname)
-            hostname_ptr = FFI::MemoryPointer.new(hostname)
+          def self.getffiaddrinfo(hostname) # rubocop:disable  Metrics/AbcSize
+            ret = FFI::MemoryPointer.new(Facter::Util::Resolvers::Ffi::AddrInfo)
 
             hints = Facter::Util::Resolvers::Ffi::AddrInfo.new
             hints[:ai_family] = Socket::AF_UNSPEC
             hints[:ai_socketype] = Socket::SOCK_STREAM
             hints[:ai_flags] = Socket::AI_CANONNAME
 
-            res = Hostname.getaddrinfo(hostname_ptr, FFI::Pointer::NULL, hints.to_ptr, FFI::Pointer::NULL)
-            return if res != 0 || res != EAI_NONAME
+            res = Hostname.getaddrinfo(hostname, '', hints.to_ptr, ret)
+            log = Log.new(self)
+            log.debug("FFI Getaddrinfo finished with exit status: #{res}")
+            return if res != 0 && res != EAI_NONAME
```

Comment:
if `res == 0`, then from that point forward we own the native memory returned by `getaddrinfo` and we need to free it by calling `freeaddrinfo`, preferably in an ensure block:

> The freeaddrinfo() function frees the memory that was allocated for the dynamically allocated linked list res.

Reply:
We could log here a debug/warning message, similar to facter 3